uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php

```diff
@@ -468,21 +468,48 @@ public function upgrade_1017()
 
     return false;
   }
-    
+
+  /**
+   * Make sure that the Tasks and Assignments main menu item
+   * (and subsequently the Dashboard menu item) are restricted to only the users
+   * with 'access Tasks and Assignments' permission
+   */
+  public function upgrade_1018() {
+    $taNavigation = new CRM_Core_BAO_Navigation();
+    $taNavigation->name = 'tasksassignments';
+    $taNavigation->find(true);
+
+    if($taNavigation->id && !$taNavigation->permission) {
+      $navigation = new CRM_Core_BAO_Navigation();
+      $foo = array(
+        'id' => $taNavigation->id,
+        'permission' => 'access Tasks and Assignments',
+        'separator'  => 1,
```

Comment:
Unfortunately there is, if you don't define them (in 4.5) it defaults them to `false`, see [here](https://github.com/civicrm/civicrm-core/blob/4.5/CRM/Core/BAO/Navigation.php#L94-L95)
```php
$params['is_active'] = CRM_Utils_Array::value('is_active', $params, FALSE);
$params['has_separator'] = CRM_Utils_Array::value('has_separator', $params, FALSE);
```